src/main/java/dev/morling/onebrc/WeatherStationFactory.java

```diff
@@ -0,0 +1,144 @@
+        var bigName = new StringBuilder(1 << 20);
+
+        try (var rows = new BufferedReader(new FileReader("data/weather_stations.csv"));) {
+            skipComments(rows);
+            while (true) {
+                var row = rows.readLine();
+                if (row == null) {
+                    break;
+                }
+                bigName.append(row, 0, row.indexOf(';'));
+            }
+        }
+
+        final var weatherStations = new ArrayList<WeatherStation>();
+        final var names = new HashSet<String>();
+        var minLen = Integer.MAX_VALUE;
+        var maxLen = Integer.MIN_VALUE;
+        try (var rows = new BufferedReader(new FileReader("data/weather_stations.csv"))) {
+            skipComments(rows);
+            final var rnd = new FastRandom(seed);
+            final var nameSource = new StringReader(bigName.toString());
+            final var buf = new char[MAX_NAME_LEN];
+            final double yOffset = 4;
+            final double factor = 2500;
+            final double xOffset = 0.372;
+            final double power = 7;
+            for (int i = 0; i < KEYSET_SIZE; i++) {
+                var row = rows.readLine();
+                if (row == null) {
+                    break;
+                }
+                // Use a 7th-order curve to simulate the name length distribution.
+                // It gives us mostly short names, but with large outliers.
+                var nameLen = (int) (yOffset + factor * Math.pow(rnd.nextDouble() - xOffset, power));
+                var count = nameSource.read(buf, 0, nameLen);
+                if (count == -1) {
+                    // TODO: Reset nameSource and read instead of throwing?
+                    throw new Exception("Name source exhausted");
+                }
+
+                var nameBuf = new StringBuilder(nameLen);
+                nameBuf.append(buf, 0, nameLen);
+
+                if (Character.isWhitespace(nameBuf.charAt(0))) {
+                    nameBuf.setCharAt(0, readNonSpace(nameSource));
+                }
+                if (Character.isWhitespace(nameBuf.charAt(nameBuf.length() - 1))) {
+                    nameBuf.setCharAt(nameBuf.length() - 1, readNonSpace(nameSource));
+                }
+
+                var name = nameBuf.toString();
+                int actualLen = name.getBytes(StandardCharsets.UTF_8).length;
+                while (actualLen > 100) {
+                    nameBuf.deleteCharAt(nameBuf.length() - 1);
+                    if (Character.isWhitespace(nameBuf.charAt(nameBuf.length() - 1))) {
+                        nameBuf.setCharAt(nameBuf.length() - 1, readNonSpace(nameSource));
+                    }
+                    name = nameBuf.toString();
+                    actualLen = name.getBytes(StandardCharsets.UTF_8).length;
+                }
+
+                while (names.contains(name)) {
+                    nameBuf.setCharAt(rnd.nextInt(nameBuf.length()), readNonSpace(nameSource));
```

Comment:
Here's the trouble: this change may result in the encoded string exceeding 100 bytes.

Reply:
Ah, right. Characters can be > 1 byte. Will fix

Reply:
Fixed. Now loops `while (names.contains(name) || nameByteLen > 100)` and takes the appropriate action.